### PR TITLE
Add upgrade test

### LINKED ISSUE #1664: Set up upgrade tests for knative-gcp
**Problem**
Currently the knative/eventing repo runs as part of their CI, [upgrade tests](https://github.com/knative/eventing/blob/master/test/e2e-upgrade-tests.sh).
Knative-gcp repo, however, does not have such tests. 

**[Persona:](https://github.com/knative/eventing/blob/master/docs/personas.md)** 

developer

**Exit Criteria**
Upgrade tests are setup for relevant resources defined in this repo.

**Time Estimate (optional):** 
~2w

**Additional context (optional)**
The upgrade tests should be similar to the ones in knative/eventing repo. They should re-use as much as possible the existing scripts.

The upgrade tests should also have a setup similar to existing ones in the knative-gcp repo (e.g., [e2e-tests](https://github.com/google/knative-gcp/blob/master/test/e2e-tests.sh)). Specifically as related to how versions of knative/eventing are obtained.

Other useful pointers:
Recent PR enabling some upgrade tests: https://github.com/knative/eventing/pull/3927
Older PRs adding upgrade tests: https://github.com/knative/eventing/pull/3262, https://github.com/knative/eventing/pull/3110

### PR DESCRIPTION
**Pull request by zhongduo**
Fixes #1664 

## Proposed Changes

- Add upgrade test similar to knative eventing
-
-

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

File: test/upgrade/prober/verify.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	report := fetchReceiverReport(address, p.log)
+	p.log.Infof("Fetched receiver report. Events propagated: %v. "+
+		"State: %v", report.Events, report.State)
+	if report.State == "active" {
+		panic(errors.New("report fetched to early, receiver is in active state"))
+	}
+	errs := make([]error, 0)
+	for _, t := range report.Thrown {
+		if strings.Contains(t, "should be received once") {
+			p.log.Warn("Duplicate events: ", t)
```
Comment (nlopezgi) [inline test/upgrade/prober/verify.go:48]: Should this be an error? Not sure what the purpose of the warning or what someone who sees it should do?

**Review by nlopezgi — COMMENTED** [review]
few minor nits

File: test/e2e-common.sh [hunk 2]
```diff
@@ -144,3 +144,26 @@ function dump_extra_cluster_state() {
...
+      return 1
+    fi
+
+    sleep 1
+
+    # Decrease the timeout of one
+    ((timeout--))
+  done
+  return 0
+}
```
[the remark this answers is not in the document]
Comment (zhongduo) [reply test/e2e-common.sh]: done

File: test/lib.sh [hunk 3]
```diff
@@ -38,21 +38,87 @@ function start_knative_gcp() {
...
+    ko apply --strict -f ${kne_config} || return 1
+  else
+    kubectl apply -f $kne_config || return 1
+  fi
+  ko apply --strict -f ${CLOUD_RUN_EVENTS_ISTIO_CONFIG}|| return 1
   wait_until_pods_running cloud-run-events || return 1
 }
 
+latest_version() {
+  #local semver=$(git describe --match "v[0-9]*" --abbrev=0)
```
[the remark this answers is not in the document]
Comment (zhongduo) [reply test/lib.sh:67]: Done as suggested.

File: test/lib.sh [hunk 4]
```diff
@@ -38,21 +38,87 @@ function start_knative_gcp() {
...
+  #local major_minor=$(echo "$semver" | cut -d. -f1-2)
+
+  # Get the latest patch release for the major minor
+  #git tag -l "${major_minor}*" | sort -r --version-sort | head -n1
+  
+  # FIXME: currently we don't have the right version tag in master,
```
[the remark this answers is not in the document]
Comment (zhongduo) [reply test/lib.sh]: done as suggested.

File: test/upgrade/prober/verify.go (hunk 1 above)
Comment (zhongduo) [reply test/upgrade/prober/verify.go:48]: Duplicate events are not really errors as our guarantee is "at least once". So we just log it here. If later we see this is a concerning problem, we can make it error.

**Pull request merged**